.github/workflows/check-examples.yaml

```diff
@@ -0,0 +1,196 @@
+          astartectl realm-management interfaces sync $GITHUB_WORKSPACE/examples/example_device/interfaces/*.json --non-interactive
+          astartectl realm-management interfaces ls
+      - name: Setup the configuration file for the datastream example
+        working-directory: ./examples/datastreams
+        run: |
+          echo "REALM = \"test\"" > config.toml
+          echo "PAIRING_URL = \"https://api.autotest.astarte-platform.org/pairing\"" >> config.toml
+          DEVICE_ID=$(astartectl utils device-id generate-random)
+          echo "DEVICE_ID = \"$DEVICE_ID\"" >> config.toml
+          CREDENTIALS_SECRET=$(astartectl pairing agent register --compact-output -- "$DEVICE_ID")
+          echo "CREDENTIALS_SECRET = \"$CREDENTIALS_SECRET\"" >> config.toml
+          cat config.toml
+      - name: Setup the configuration file for the event_listener example
+        working-directory: ./examples/event_listener
+        run: |
+          echo "REALM = \"test\"" > config.toml
+          echo "PAIRING_URL = \"https://api.autotest.astarte-platform.org/pairing\"" >> config.toml
+          DEVICE_ID=$(astartectl utils device-id generate-random)
+          echo "DEVICE_ID = \"$DEVICE_ID\"" >> config.toml
+          CREDENTIALS_SECRET=$(astartectl pairing agent register --compact-output -- "$DEVICE_ID")
+          echo "CREDENTIALS_SECRET = \"$CREDENTIALS_SECRET\"" >> config.toml
+          cat config.toml
+      - name: Setup the configuration file for the example device example
+        working-directory: ./examples/example_device
+        run: |
+          echo "REALM = \"test\"" > config.toml
+          echo "PAIRING_URL = \"https://api.autotest.astarte-platform.org/pairing\"" >> config.toml
+          DEVICE_ID=$(astartectl utils device-id generate-random)
+          echo "DEVICE_ID = \"$DEVICE_ID\"" >> config.toml
+          CREDENTIALS_SECRET=$(astartectl pairing agent register --compact-output -- "$DEVICE_ID")
+          echo "CREDENTIALS_SECRET = \"$CREDENTIALS_SECRET\"" >> config.toml
+          cat config.toml
+      - name: Install the Astarte device Python module
+        run: |
+          python3 -m pip install --upgrade pip
+          python3 -m pip install -e .
+      - name: Execute the datastream example
+        working-directory: ./examples/datastreams
+        run: |
+          export REQUESTS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
+          python3 ./main.py -d 10
+      - name: Execute the event_listener example
+        working-directory: ./examples/event_listener
+        run: |
+          export REQUESTS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
+          python3 ./event_listener.py -d 10
+      - name: Execute the example_device example
+        working-directory: ./examples/example_device
+        run: |
+          export REQUESTS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
+          python3 ./example_device.py -d 10
+  check-examples-grpc:
```

Comment:
Both workflows are testing examples but since they are getting lengthy we could still separate them. I don't think this would be a problem